README.md

```diff
@@ -43,67 +41,130 @@ import Vosk from 'react-native-vosk';
+    const resultEvent = vosk.onResult((res) => {
+      console.log('A onResult event has been caught: ' + res);
     });
 
     // Don't forget to call resultEvent.remove(); to delete the listener
-}).catch(e => {
+  })
+  .catch((e) => {
     console.error(e);
-})
-
+  });
 ```
 
 Note that `start()` method will ask for audio record permission.
 
-[Complete example...](https://github.com/riderodd/react-native-vosk/blob/main/example/src/App.tsx)
+[See complete example...](https://github.com/riderodd/react-native-vosk/blob/main/example/src/App.tsx)
 
 ### Methods
 
 | Method | Argument | Return | Description |
 |---|---|---|---|
-| `loadModel` | `path: string` | `Promise` | Loads the voice model used for recognition, it is required before using start method |
-| `start` | `grammar: string[]` or `none` | `Promise` | Starts the voice recognition and returns the recognized text as a promised string, you can recognize specific words using the `grammar` argument (ex: ["left", "right"]) according to kaldi's documentation |
-| `stop` | `none` | `none` | Stops the recognition |
+| `loadModel` | `path: string` | `Promise<void>` | Loads the voice model used for recognition, it is required before using start method. |
+| `start` | `options: VoskOptions` or `none` | `Promise<void>` | Starts the recognizer, an `onResult()` event will be fired. |
+| `setGrammar` | `grammar: string[]` or `none` | `Promise<void>` | Reconfigures the recognizer to use grammar. |
+| `stop` | `none` | `none` | Stops the recognizer. Listener should receive final result if there is any. |
+| `unload` | `none` | `none` | Unloads the model, also stops the recognizer. |
+
+### Types
+
+| VoskOptions | Type | Required | Description |
+|---|---|---|---|
+| `grammar` | `string[]` | No | Set of phrases the recognizer will seek on which is the closest one from the record, add `"[unk]"` to the set to recognize phrases striclty. |
+| `timeout` | `int` | No | Timeout in milliseconds to listen. |
 
 ### Events 
 
 | Method | Promise return | Description |
 |---|---|---|
-| `onResult` | The recognized word as a `string` | Triggers on voice recognition result |
-| `onFinalResult` | The recognized word as a `string` | Triggers if stopped using `stop()` method |
-| `onError` | The error that occured as a `string` or `exception` | Triggers if an error occured |
-| `onTimeout` | "timeout" `string` | Triggers on timeout |
+| `onPartialResult` | The recognized word as a `string` | Called when partial recognition result is available.|
+| `onResult` | The recognized word as a `string` | Called after silence occured. |
+| `onFinalResult` | The recognized word as a `string` | Called after stream end, like a `stop()` call |
+| `onError` | The error that occured as a `string` or `exception` | Called when an error occurs |
+| `onTimeout` | `void` | Called after timeout expired |
+
+### Examples
+
+#### Default
+
+```js
+const start = () => vosk.start();
+
+useEffect(() => {
```

Comment:
useEffect can be misinterpreted here, I think we should show a more agnostic example, don't you ?